manifests/client.pp

```diff
@@ -9,10 +9,10 @@
 #   is repository dependent.
 #
 class mongodb::client (
-  $ensure       = $mongodb::params::package_ensure_client,
-  $package_name = $mongodb::params::client_package_name,
+  Variant[Boolean, String] $ensure = $mongodb::params::package_ensure_client,
+  Optional[String] $package_name   = $mongodb::params::client_package_name,
 ) inherits mongodb::params {
-  anchor { '::mongodb::client::start': }
-  -> class { '::mongodb::client::install': }
-  -> anchor { '::mongodb::client::end': }
+  anchor { 'mongodb::client::start': }
```

Comment:
Maybe replace the anchor as well with a contain?

Reply:
Yeah, I want to do the structural changes in another PR (if you want to start working on one, that would be great). But might be good to get the acceptance tests in a better place first? Right now they're a total disaster, maybe worse than when we started.